avocado/core/runners/avocado_instrumented.py

```diff
@@ -77,41 +114,41 @@ def run(self):
 
         time_started = time.monotonic()
         yield self.prepare_status('started')
-
-        # Waiting for early status
-        while queue.empty():
-            time.sleep(nrunner.RUNNER_RUN_CHECK_INTERVAL)
-
-        early_status = queue.get()
-        timeout = float(early_status.get('timeout') or self.DEFAULT_TIMEOUT)
-        interrupted = False
+        early_status = {}
+        timeout = float(self.DEFAULT_TIMEOUT)
         most_current_execution_state_time = None
-        while queue.empty():
+        while True:
             time.sleep(nrunner.RUNNER_RUN_CHECK_INTERVAL)
             now = time.monotonic()
-            if most_current_execution_state_time is not None:
-                next_execution_state_mark = (most_current_execution_state_time +
-                                             nrunner.RUNNER_RUN_STATUS_INTERVAL)
-            if (most_current_execution_state_time is None or
-                    now > next_execution_state_mark):
-                most_current_execution_state_time = now
-                yield self.prepare_status('running')
-            if (now - time_started) > timeout:
-                process.terminate()
-                interrupted = True
-                break
-        if interrupted:
-            status = early_status
-            status['result'] = 'interrupted'
-            status['status'] = 'finished'
-            if 'name' in status:
-                del status['name']
-            if 'time_start' in status:
-                del status['time_start']
-        else:
-            status = queue.get()
-        status['time'] = time.monotonic()
-        yield status
+            if queue.empty():
+                if most_current_execution_state_time is not None:
+                    next_execution_state_mark = (most_current_execution_state_time +
+                                                 nrunner.RUNNER_RUN_STATUS_INTERVAL)
+                if (most_current_execution_state_time is None or
+                        now > next_execution_state_mark):
+                    most_current_execution_state_time = now
+                    yield self.prepare_status('running')
+                if (now - time_started) > timeout:
+                    process.terminate()
+                    status = early_status
+                    status['result'] = 'interrupted'
+                    if 'name' in status:
+                        del status['name']
+                    if 'time_start' in status:
+                        del status['time_start']
+                    yield self.prepare_status('finished', status)
+                    break
+            else:
+                status = queue.get()
```

Comment:
I'm concerned about the names here. In the queue, are we storing status or messages? I have an impression that `message = queue.get()` makes more sense.

Reply:
or maybe status_message.